### PR TITLE
Fixed setup.py when missing libjpeg

### PR DESCRIPTION
**Pull request by vfdev-5**
This PR fixes the error when installing torchvision from source on a system without libjpeg (e.g. GH Codespaces)

```
/workspaces/vision/torchvision/csrc/io/image/cpu/decode_jpeg.cpp:159:10: error: 'JPEG_LIB_VERSION' was not declared in this scope; did you mean 'PSTL_VERSION'?
  159 |   return JPEG_LIB_VERSION;
      |          ^~~~~~~~~~~~~~~~
      |          PSTL_VERSION
error: command '/usr/bin/gcc' failed with exit code 1
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,9 +82,10 @@ Torchvision currently supports the following image backends:
 - [libjpeg](http://ijg.org/) - can be installed via conda `conda install jpeg` or any of the package managers for
   debian-based and RHEL-based Linux distributions. [libjpeg-turbo](https://libjpeg-turbo.org/) can be used as well.
 
-**Notes:** `libpng` and `libjpeg` must be available at compilation time in order to be available. Make sure that it is
-available on the standard library locations, otherwise, add the include and library paths in the environment variables
-`TORCHVISION_INCLUDE` and `TORCHVISION_LIBRARY`, respectively.
+**Notes:** `libpng` and `libjpeg` are optional dependencies. If any of them is available on the system, 
+torchvision will provide encoding/decoding image functionalities from `torchvision.io.image`. 
+When building torchvision from source, `libpng` and `libjpeg` can be found on the standard library locations.
```
Comment (NicolasHug) [inline README.md:87]: I don't think we should say that. It depends on whether the user installed them.

File: README.md [hunk 2]
```diff
@@ -82,9 +82,10 @@ Torchvision currently supports the following image backends:
 - [libjpeg](http://ijg.org/) - can be installed via conda `conda install jpeg` or any of the package managers for
   debian-based and RHEL-based Linux distributions. [libjpeg-turbo](https://libjpeg-turbo.org/) can be used as well.
 
-**Notes:** `libpng` and `libjpeg` must be available at compilation time in order to be available. Make sure that it is
-available on the standard library locations, otherwise, add the include and library paths in the environment variables
-`TORCHVISION_INCLUDE` and `TORCHVISION_LIBRARY`, respectively.
+**Notes:** `libpng` and `libjpeg` are optional dependencies. If any of them is available on the system,
```
Comment (NicolasHug) [inline README.md:85]: It's good to clarify that libjpeg and libpng are optional, but I don't think we need to remove that sentence:

> `libpng` and `libjpeg` must be available at compilation time in order to be available.

**Review by NicolasHug — APPROVED** [review]
Thanks @vfdev-5 , LGTM if green with a few comments. Maybe we should just merge it and completely re-write that part of the readme in another PR. We should remove mentions of accimage and also be clear that jpeg-turbo is preferred now.

File: README.md (hunk 2 above)
Comment (vfdev-5) [reply README.md:85]: Sorry, I do not understand what you ask. Libs are optional and this sentence "libpng and libjpeg must be available at compilation time in order to be available." does not make sense (without speaking that it is unclear what it is exactly "in order be available")

Comment (NicolasHug) [reply README.md:85]: I think the sentence means

> libjpeg and libpng need to be available at compile time when you build from source.

We don't need to remove it

File: README.md (hunk 1 above)
Comment (NicolasHug) [reply README.md:87]: We don't know whether those libraries can be doung on the standard library locations. If users didn't install them, then they won't be found there.

Comment (vfdev-5) [reply README.md:87]: OK, I think I need to rephrase the sentence I wrote: "`libpng` and `libjpeg` are first searched on the standard library locations"

**Comment by NicolasHug** [comment]
Thanks for the fix @vfdev-5 , I'll submit a PR soon to rework our README a bit, as other parts of it also need an update

**Pull request merged**